Scrollable Navigation (#2577)
BBC-archive/psammead: packages/components/psammead-navigation/README.md

The `@bbc/psammead-navigation` package is a set of two components, `NavigationUl` and `NavigationLi`. They use `ul` and `li` HTML elements respectively.

## Exports

> `/swipeable` - Makes the navigation to be swipeable under 600px.

Reviewer: Re name to 'scrollable' please - we aren't adding any swiping gestures?

Reviewer: Same goes for anywhere else we've used the word swipeable...

| -------- | ---- | -------- | ------- | ------- |
| children | node | Yes      | N/A     | `<NavigationUl><NavigationLi url="/" script={latin} active="true">Home</NavigationLi><NavigationLi url="/sport" script={latin}>{Sport}</NavigationLi></NavigationUl>` |
| dir      | string  | No       | `ltr`   | `rtl` |
| isSwipeable | boolean | No | `false` | `true` |

### AmpSwipeableNavigation

<!-- prettier-ignore -->
| Argument | Type | Required | Default | Example |
| -------- | ---- | -------- | ------- | ------- |
| children | node | Yes      | N/A     | `<NavigationUl><NavigationLi url="/" script={latin} active="true">Home</NavigationLi><NavigationLi url="/sport" script={latin}>{Sport}</NavigationLi></NavigationUl>` |
| dir      | string  | No       | `ltr`   | `rtl` |

## Usage

```jsx
import React from 'react';
import { CanonicalSwipeableNavigation } from '@bbc/psammead-navigation/swipeable-navigation';
import Navigation, {
  NavigationUl,
  NavigationLi,
} from '@bbc/psammead-navigation';
import { latin } from '@bbc/gel-foundations/scripts';

<Navigation>
  <CanonicalSwipeableNavigation>
    <NavigationUl>
      <NavigationLi
        url="/"
        script={latin}

>         active="true"

Reviewer 2: ```suggestion
        active
```

        currentPageText="Current Page"
        service="news"
      >
        Home
      </NavigationLi>
      <NavigationLi url="/sport" script={latin} service="news">
        {Sport}
      </NavigationLi>
      <NavigationLi url="/weather" script={latin} service="news">
        {Weather}
      </NavigationLi>
    </NavigationUl>
  </CanonicalSwipeableNavigation>
</Navigation>;
```

### When to use this component

The `Navigation` is designed to show a navigation bar on `index` pages, which will show all sections on a site. If there are too many items to fit on one line, the items will wrap to the next lines.

> On the other hand, with `CanonicalSwipeableNavigation` or `AmpSwipeableNavigation` we can make the list to remain on one line and to be horizontally swipeable to allow access to further links, under 600px.

Reviewer: 'horizontally swipeable to' should be 'horizontally scrollable to'

### Accessibility notes

The Navigation has a [`navigation` landmark](https://www.w3.org/TR/wai-aria-practices/examples/landmarks/navigation.html) to provide a way to identify links that are intended to be used for navigation.

We have added the role `list` and `listitem` to the `NavigationUl` and `NavigationList` respectively, due to a VoiceOver bug to reinstate the list semantics.

We have also added visually hidden text to let the user know which item in the Navigation is the current page. Note the use of visually hidden text here is due to lack of support at this time for the aria-current page attribute. Also note the use of `role="text"` to stop splitting in VoiceOver.

> In the screen reader UX only the menu button and its content should be available to assistive technology. The exposed swipeable navigation should be hidden. To achieve this we add `aria-hidden:true` to the exposed swipeable navigation so that this is not expose to these users and also add `tabindex=-1` to the links contained within this to remove them from the tab order.

Reviewer 2: Can we replace the word `exposed` with `visible`?

Author: I think `exposed` is more appropriate in this context because the navigation will be still visible, but not exposed to screen readers.

Reviewer 2: Aaah ok. I was misinterpreting what was meant by the statement because of the use of `exposed` and `hidden` in the same sentence. Would it be better to restructure the sentence a little.

```suggestion
In the screen reader UX only the menu button and its content should be available to assistive technology. The swipeable navigation is exposed to assistive technologies and  should be hidden. To achieve this we add `aria-hidden=true` to the swipeable navigation. We also add `tabindex=-1` to the links contained within this to remove them from the tab order.
```

Author: I renamed it in a different way, hope this is more clear.

> In the screen reader UX only the menu button and its content should be available to assistive technology, for which reason the swipeable navigation should be hidden. To achieve this we add `aria-hidden:true` to the exposed swipeable navigation so that this is not visible to these users and also add `tabindex=-1` to the links contained within this to remove them from the tab order.

Reviewer: In the screen reader UX only the menu button and its content should be available to assistive technology, **meaning the scrollable navigation will be hidden**. To achieve this we add `aria-hidden:true` to the exposed **scrollable** navigation so that this is not visible to these users and also add `tabindex=-1` to the links contained within this to remove them from the tab order.
